### PR TITLE
stdenv: Two small cleanups

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Description of changes

1. `common-path.nix`: move to `stdenv/generic/`.

2. Simplify dispatching to choice of bootstrapping / stdenv.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/stdenv/default.nix [hunk 1]
```diff
@@ -38,28 +38,10 @@ let
...
-    armv7l-linux = stagesLinux;
-    armv7r-linux = stagesLinux;
-    armv7m-linux = stagesLinux;
-    armv8a-linux = stagesLinux;
-    armv8r-linux = stagesLinux;
-    armv8m-linux = stagesLinux;
-    aarch64-linux = stagesLinux;
-    mipsel-linux = stagesLinux;
-    mips64el-linux = stagesLinux;
-    powerpc-linux = /* stagesLinux */ stagesNative;
```
Comment (Mindavi) [inline pkgs/stdenv/default.nix:57]: You missed this one special case :)

Comment (Ericson2314) [reply pkgs/stdenv/default.nix:57]: I mentioned it in the commit message. I don't think special case ought to exist, and I don't think anyone is using `powerpc-linux` as opposed to the newer Power arches anyways

Comment (Mindavi) [reply pkgs/stdenv/default.nix:57]: Sorry about that, makes sense.

**Review by Mindavi — APPROVED** [review]

**Review by roberth — APPROVED** [review]
OT: Would be nice to use `common-path` to define a better `stdenvNoCC` called `commonEnv` perhaps?

**Comment by Ericson2314** [comment]
> OT: Would be nice to use common-path to define a better stdenvNoCC called commonEnv perhaps?

If you mean it would be better to build `stdenv` from `stdenvNoCC` by adding the C compiler than the reverse like today where we remove it, then yes I agree.

**Pull request merged**